src/cargo/core/compiler/fingerprint/dep_info.rs

```diff
@@ -129,6 +152,16 @@ impl EncodedDepInfo {
         }
         return Some(EncodedDepInfo { files, env });
 
+        fn read_version(bytes: &mut &[u8]) -> Option<u8> {
+            let _size = read_usize(bytes)?;
+            let path_type = read_u8(bytes)?;
+            if path_type != u8::MAX {
+                // Old depinfo. Give up parsing it.
+                return None;
+            }
```

Comment:
I wonder if this should be split out into a `read_header` or `read_magic_marker` or something.  I found it confusing when I saw the doc comment and then immediately saw `read_version`.  I was then going to comment in `read_version` about how it was behaving